Swift/FullTextFunction.swift

```diff
@@ -28,17 +28,55 @@ public final class FullTextFunction {
     ///
     /// - Parameter indexName: The index name.
     /// - Returns: The full-text rank function.
+    @available(*, deprecated, message: "Use FullTextFunction.rank(withIndex:) instead.")
     public static func rank(_ indexName: String) -> ExpressionProtocol {
-        return QueryExpression(CBLQueryFullTextFunction.rank(indexName))
+        let index = Expression.fullTextIndex(indexName)
+        guard let i = index as? FullTextIndexExpression else {
+            fatalError("This scenario shouldn't happen! IndexExpProtocol only implemented on FullTextIndexExpression")
+        }
+        return QueryExpression(CBLQueryFullTextFunction.rank(withIndex: i.toImpl()))
     }
     
     /// Creates a full-text match expression with the given full-text index name and the query text
     ///
     /// - Parameter indexName: The index name.
     /// - Parameter query: The query string.
     /// - Returns: The full-text match function expression.
+    @available(*, deprecated, message: "Use FullTextFunction.match(withIndex:query) instead.")
     public static func match(indexName: String, query: String) -> ExpressionProtocol {
-        return QueryExpression(CBLQueryFullTextFunction.match(withIndexName: indexName,
+        let index = Expression.fullTextIndex(indexName)
+        guard let i = index as? FullTextIndexExpression else {
```

Comment:
Should be just `return match(Expression.fullTextIndex(indexName), query: query)`?

Reply:
cleaner!